subprojects/logging/src/main/java/org/gradle/internal/logging/LoggingConfigurationBuildOptions.java

```diff
@@ -170,4 +177,31 @@ public void applyTo(String value, LoggingConfiguration settings, Origin origin)
             }
         }
     }
+
+    public static class WarningsOption extends ListBuildOption<LoggingConfiguration> {
+        public static final String LONG_OPTION = "warnings";
+        public static final String GRADLE_PROPERTY = "org.gradle.warnings";
+
+        public WarningsOption() {
+            super(GRADLE_PROPERTY, CommandLineOptionConfiguration.create(LONG_OPTION, "Specifies which type of warnings to generate. Values are 'all', 'summary' (default) or 'no'"));
+        }
+
+        @Override
+        public void applyTo(List<String> values, LoggingConfiguration settings, final Origin origin) {
```

Comment:
:x: I'd think that we'll want to have some unit testing for this logic.

Reply:
Done.